server/odc-service/src/main/java/com/oceanbase/odc/service/session/model/AsyncExecuteContext.java

```diff
@@ -0,0 +1,140 @@
+import static com.oceanbase.odc.core.session.ConnectionSessionConstants.CONSOLE_DS_KEY;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Future;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import org.springframework.jdbc.core.StatementCallback;
+
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.common.util.VersionUtils;
+import com.oceanbase.odc.core.session.ConnectionSession;
+import com.oceanbase.odc.core.session.ConnectionSessionUtil;
+import com.oceanbase.odc.core.sql.execute.model.JdbcGeneralResult;
+import com.oceanbase.odc.core.sql.execute.model.SqlTuple;
+import com.oceanbase.odc.core.sql.util.OBUtils;
+import com.oceanbase.odc.service.iam.model.User;
+import com.oceanbase.odc.service.iam.util.SecurityContextUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author liuyizhuo.lyz
+ * @date 2024/4/15
+ */
+@Getter
+@Slf4j
+public class AsyncExecuteContext<T> {
+    public static final String SHOW_TABLE_COLUMN_INFO = "SHOW_TABLE_COLUMN_INFO";
+
+    private final Function<JdbcGeneralResult, T> mapper;
+    private final ConnectionSession session;
+    private final List<SqlTuple> sqlTuples;
+    private final Queue<T> results = new ConcurrentLinkedQueue<>();
+    private final List<BiConsumer<ConnectionSession, String>> traceIdHooks = new ArrayList<>();
+    private final User user;
+    private final Long waitTimeMillis;
+    private final List<String> sessionIds;
+
+    @Setter
+    private Future<List<JdbcGeneralResult>> future;
+    private String currentTraceId;
+    @Setter
+    private String sql;
+    private int count = 0;
+
+    public AsyncExecuteContext(ConnectionSession session, List<SqlTuple> sqlTuples,
+            Function<JdbcGeneralResult, T> mapper, User user, Long waitTimeMillis) {
+        this.session = session;
+        this.sqlTuples = sqlTuples;
+        this.mapper = mapper;
+        this.user = user;
+        this.waitTimeMillis = waitTimeMillis;
+        this.sessionIds = initSessionIds(session);
+    }
+
+    public int getTotal() {
+        return sqlTuples.size();
+    }
+
+    public List<T> getResults() {
+        List<T> copiedResults = new ArrayList<>();
+        while (!results.isEmpty()) {
+            copiedResults.add(results.poll());
+        }
+        return copiedResults;
+    }
+
+    public void setCurrentTraceId(String traceId) {
+        currentTraceId = traceId;
+        for (BiConsumer<ConnectionSession, String> hook : traceIdHooks) {
+            hook.accept(session, traceId);
+        }
+    }
+
+    public void onQueryStart(String sql) {
+        count++;
+        this.sql = sql;
+    }
+
+    public void onQueryExecuting() {
+        String traceId = session.getSyncJdbcExecutor(BACKEND_DS_KEY)
```

Comment:
OB specific logic should not exist here~

Reply:
done, move it into `OBExecutionListener`